### PR TITLE
Move to the new Google Maps API.

### PR DESCRIPTION
**Pull request by lawrencehoffman**
Update to new api. Have tested this code on Void, Ubuntu, and Arch Linux (I have no Windoz or OSX machines running at this time so those have not been tested, but should work)

Note: The new api requires a key, to set the key edit the file server/google_api.py and paste your API key in the variable named GOOGLE_API_KEY.

Note: This update attempts to make as few changes as possible, but to ease future maintenance does change the data structure returned from parsers from a list to a dictionary, as the new maps api requires a json post.

### CODE REVIEW

**Comment by lanmaster53** [comment]
What's the impact on the agent side? Anything? Or do they continue to report to the API as normal? This is probably obvious in the code, but I didn't have time to look through it at the moment.

File: .gitignore [hunk 1]
```diff
@@ -3,3 +3,4 @@
 venv/
 agents/
 server_old/
+.idea*
```
[the remark this answers is not in the document]
Comment (lawrencehoffman) [reply .gitignore]: Sorry about that. Will remove and amend.

File: server/honeybadger/views.py [hunk 2]
```diff
@@ -326,6 +324,7 @@ def get_json(url):
         return None
     return jsondata
 
+
```
[the remark this answers is not in the document]
Comment (lawrencehoffman) [reply server/honeybadger/views.py]: Auto added by pep8 plugin. Will remove.

File: server/honeybadger/views.py [hunk 3]
```diff
@@ -271,15 +272,12 @@ def api_beacon(target, agent):
                 aps = parse_netsh(content)
             elif re.search('^linux', os.lower()):
                 aps = parse_iwlist(content)
-            # handle recognized data
-            if aps:
-                url = 'https://maps.googleapis.com/maps/api/browserlocation/json?browser=firefox&sensor=true'
-                query = '&wifi=mac:{}|ssid:{}|ss:{}'
-                for ap in aps:
-                    url += query.format(ap[1], ap[0], ap[2])
-                jsondata = get_json(url[:1900])
```
[the remark this answers is not in the document]
Comment (lawrencehoffman) [reply server/honeybadger/views.py]: I did copy over a lot of the logging in the get_json function. However, as Google now requires an API key (which becomes part of the URL) I did not log the url used for the API as I didn't think it was appropriate to have the API key leaking into the logs.

File: server/honeybadger/views.py [hunk 4]
```diff
@@ -229,7 +230,7 @@ def api_beacons():
 
 # agent api views
 
-@app.route('/api/beacon/<target>/<agent>')
+@app.route('/api/beacon/<target>/<agent>', methods=['GET','POST'])
```
[the remark this answers is not in the document]
Comment (lawrencehoffman) [reply server/honeybadger/views.py]: Perhaps ignorance? Neither python, flask, nor HTTP are subjects I'd consider myself having obtained mastery over. I know that by default flask listens for GET and thereby HEAD, but worried about that old warning of [not relying on GET Requests over 255 bytes](https://www.w3.org/Protocols/rfc2616/rfc2616-sec3.html#sec3.2.1) though it probably no longer applies. Is there a good reason to disallow post? I honestly don't know... Just from the little I do know about HTTP the sending of base 64 encoded output from a local program *seemed to me* like the appropriate use case for the POST method.

**Comment by lanmaster53** [comment]
Ok... to make along story short, I went on a code refactoring binge with this project over the past 2 weeks and made a ton of changes. Some of the changes are actually things you recommended (update Google API, refactor the parsers, the ability to use POST, etc.) while others are code quality things that I've been wanting to do for a long time. I wrote this a couple years ago and I'm much more knowledgeable about the technology stack now than I was then. All that being said, I think we need to close this pull request and have you take a look at the new stuff. I think you'll enjoy the changes. Then, if you have any questions or recommendations, hit me up and we'll discuss. There were some things you recommended that I did not do for a reason and I am happy to clarify. In any case, thank you so much for your work on this. It's needed some attention for a while.

**Pull request closed without merging**